elements/lisk-p2p/src/peer_directory/new_peers.ts

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { P2PPeerInfo } from '../p2p_types';
+import { constructPeerIdFromPeerInfo, getBucket, PEER_TYPE } from '../utils';
+
+const DEFAULT_NEW_PEER_LIST_SIZE = 128;
+const DEFAULT_NEW_PEER_BUCKET_SIZE = 32;
+const MILLISECONDS_IN_ONE_DAY = 86400000; // Formula hours*minutes*seconds*milliseconds;
+const ELIGIBLE_DAYS_FOREVICTION = 30;
+
+export interface NewPeerConfig {
+	readonly newPeerListSize?: number;
+	readonly newPeerBucketSize?: number;
+	readonly secret: number;
+}
+interface NewPeerInfo {
+	readonly peerInfo: P2PPeerInfo;
+	readonly dateAdded: Date;
+}
+
+export interface AddPeerOutcome {
+	readonly success: boolean;
+	readonly evicted: boolean;
+	readonly evictedPeer?: P2PPeerInfo;
+}
+export class NewPeers {
+	private readonly _newPeerMap: Map<number, Map<string, NewPeerInfo>>;
+	private readonly _newPeerListSize: number;
+	private readonly _newPeerBucketSize: number;
+	private readonly _secret: number;
+
+	public constructor(newPeerConfig: NewPeerConfig) {
```

Comment:
`public constructor({ newPeerBucketSize, newPeerListSize, secret }: NewPeerConfig) {` using object destructuring can make the code less verbose, but I would leave this up to the standards your following.